### PR TITLE
New version of #6

### PR DESCRIPTION
**Pull request by haraldsteinlechner**

### CODE REVIEW

File: Graphics/Formats/Assimp/Anim.hsc [hunk 1]
```diff
@@ -68,13 +70,17 @@ instance Storable Animation where
...
     mTicksPerSecond  <- (#peek aiAnimation, mTicksPerSecond) p
-    mNumChannels     <- (#peek aiAnimation, mNumChannels) p
+    logPrint mTicksPerSecond
+    mNumChannels     <- (#peek aiAnimation, mNumChannels) p :: IO CUInt
+    logPrint mNumChannels
     mChannels'       <- (#peek aiAnimation, mChannels) p
-    mChannels        <- peekArray mNumChannels mChannels'
-    mNumMeshChannels <- (#peek aiAnimation, mNumMeshChannels) p
+    mChannels        <- peekArray (fromIntegral mNumChannels) mChannels'
+    mNumMeshChannels <- (#peek aiAnimation, mNumMeshChannels) p :: IO CUInt
```
Comment (joelburget) [inline Graphics/Formats/Assimp/Anim.hsc]: weird, is the annotation necessary?

**Comment by joelburget** [comment]
Cool, thanks!

File: Graphics/Formats/Assimp/Anim.hsc (hunk 1 above)
Comment (haraldsteinlechner) [inline Graphics/Formats/Assimp/Anim.hsc:82]: i just did this for choosing the right fromIntegral instances (happened a couple of times). More improvements on the way ;)

Comment (joelburget) [inline Graphics/Formats/Assimp/Anim.hsc:82]: :+1:

**Pull request merged**